opm/input/eclipse/Schedule/Schedule.hpp

```diff
@@ -227,18 +227,17 @@ namespace Opm {
 
         bool hasGroup(const std::string& groupName, std::size_t timeStep) const;
         std::vector<std::string> groupNames(const std::string& pattern, std::size_t timeStep) const;
-        std::vector<std::string> groupNames(std::size_t timeStep) const;
+        const std::vector<std::string>& groupNames(std::size_t timeStep) const;
         std::vector<std::string> groupNames(const std::string& pattern) const;
-        std::vector<std::string> groupNames() const;
-        /*
-          The restart_groups function returns a vector of groups pointers which
-          is organized as follows:
-
-            1. The number of elements is WELLDIMS::MAXGROUPS + 1
-            2. The elements are sorted according to group.insert_index().
-            3. If there are less than WELLDIMS::MAXGROUPS nullptr is used.
-            4. The very last element corresponds to the FIELD group.
-        */
+        const std::vector<std::string>& groupNames() const;
+
+        // The restart_groups function returns a vector of groups pointers which
```

Comment:
Unrelated comment: Any reason not to switch to doxygen here like elsewhere in the PR.

Reply:
> Any reason not to switch to doxygen here like elsewhere in the PR.

Not really.  I've pushed an update which adds Doxygen-style documentation to the (nearby) group-related functions.